Use buildx, set tag to current version fails if it contains `-SNAPSHOT`

<!-- Please fill out the following information to help us in analyzing the issue, but feel free to skip it if you don't have the information at hand or if it does not apply. Please remove everything which does not apply to the issue. -->

### Description
When setting the current version (which is almost always a snapshot version in Maven), the build will fail if using `buildx`.
The errormessage suggests that the tags are passed to Docker as a repository name?
`error: invalid tag "2.4.11-SNAPSHOT": repository name must be lowercase`

The same configuration works with the regular build instead of buildx.

Example output from the PR linked below:
```
[INFO] DOCKER> docker --config /home/runner/work/S3Mock/S3Mock/docker/target/docker/adobe/s3mock/docker buildx build --progress=plain --builder dmp_adobe_s3mock --platform linux/amd64,linux/arm64 --tag 2.4.11-SNAPSHOT --tag latest --tag adobe/s3mock --cache-to=type=local,dest=/home/runner/work/S3Mock/S3Mock/docker/target/docker/adobe/s3mock/cache --cache-from=type=local,src=/home/runner/work/S3Mock/S3Mock/docker/target/docker/adobe/s3mock/cache --file Dockerfile /home/runner/work/S3Mock/S3Mock/docker/target/docker/adobe/s3mock/build
[INFO] DOCKER> WARNING: No output specified for docker-container driver. Build result will only remain in the build cache. To push result image into registry use --push or to load image into docker use --load
[INFO] DOCKER> #1 [internal] booting buildkit
[INFO] DOCKER> #1 pulling image moby/buildkit:buildx-stable-1
[INFO] DOCKER> #1 pulling image moby/buildkit:buildx-stable-1 1.0s done
[INFO] DOCKER> #1 creating container buildx_buildkit_dmp_adobe_s3mock0
[INFO] DOCKER> #1 creating container buildx_buildkit_dmp_adobe_s3mock0 0.9s done
[INFO] DOCKER> #1 DONE 1.9s
[INFO] DOCKER> error: invalid tag "2.4.11-SNAPSHOT": repository name must be lowercase
[INFO] DOCKER> docker --config /home/runner/work/S3Mock/S3Mock/docker/target/docker/adobe/s3mock/docker buildx rm dmp_adobe_s3mock
[ERROR] DOCKER> Error status (1) when building
```

### Info

* docker-maven-plugin version : 0.40.0
* Maven version (`mvn -v`) :
```
mvn -V
Apache Maven 3.8.5 (3599d3414f046de2324203b78ddcf9b5e4388aa0)
Maven home: /usr/local/Cellar/maven/3.8.5/libexec
Java version: 18.0.1, vendor: Azul Systems, Inc., runtime: /Library/Java/JavaVirtualMachines/zulu-18.jdk/Contents/Home
Default locale: en_DE, platform encoding: UTF-8
OS name: "mac os x", version: "12.4", arch: "x86_64", family: "mac"
```
* Docker version :
```
docker --version
Docker version 20.10.14, build a224086
```
* If it's a bug, how to reproduce :
in the build step, configure the `tags` to contain the current project version:
```
<tags>
  <tag>${project.version}</tag>
  <tag>latest</tag>
</tags>
```

* Sample project : *[GitHub Clone URL]*
https://github.com/adobe/S3Mock/pull/617

--- a/it/buildx/pom.xml
+++ b/it/buildx/pom.xml
@@ -35,6 +35,10 @@
                     <platform>linux/arm64</platform>
                   </platforms>
                 </buildx>
+                <tags>
+                  <tag>${project.version}</tag>
+                  <tag>latest</tag>
+                </tags>
               </build>
               <run>
                 <log>

--- a/src/main/java/io/fabric8/maven/docker/service/BuildXService.java
+++ b/src/main/java/io/fabric8/maven/docker/service/BuildXService.java
@@ -8,6 +8,7 @@ import io.fabric8.maven.docker.config.BuildImageConfiguration;
 import io.fabric8.maven.docker.config.BuildXConfiguration;
 import io.fabric8.maven.docker.config.ImageConfiguration;
 import io.fabric8.maven.docker.util.EnvUtil;
+import io.fabric8.maven.docker.util.ImageName;
 import io.fabric8.maven.docker.util.Logger;
 import io.fabric8.maven.docker.util.ProjectPaths;
 import org.apache.maven.plugin.MojoExecutionException;
@@ -119,7 +120,7 @@ public class BuildXService {
         cmdLine.add(String.join(",", platforms));
         buildConfiguration.getTags().forEach(t -> {
                 cmdLine.add("--tag");
-                cmdLine.add(t);
+                cmdLine.add(new ImageName(imageConfig.getName(), t).getFullName());
             }
         );
         cmdLine.add("--tag");

--- a/src/test/java/io/fabric8/maven/docker/service/RegistryServiceTest.java
+++ b/src/test/java/io/fabric8/maven/docker/service/RegistryServiceTest.java
@@ -29,9 +29,11 @@ import org.mockito.junit.jupiter.MockitoExtension;
 
 import java.io.File;
 import java.nio.file.Path;
+import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 /**
@@ -77,6 +79,10 @@ class RegistryServiceTest {
     @Mock
     private BuildXService.Exec exec;
 
+    private static String getOsDependentBuild(Path buildPath, String docker) {
+        return buildPath.resolve(docker).toString().replace('/', File.separatorChar);
+    }
+
     @BeforeEach
     void setup() {
         BuildXService buildXService = new BuildXService(docker, dockerAssemblyManager, logger, exec);
@@ -92,7 +98,7 @@ class RegistryServiceTest {
     }
 
     @ParameterizedTest
-    @ValueSource(booleans = {false, true})
+    @ValueSource(booleans = { false, true })
     void pullImagePullPolicyAlways(boolean hasImage) throws Exception {
         givenAnImage();
         givenAnImageConfiguration("myregistry.com/user/app:1.0.1");
@@ -103,7 +109,7 @@ class RegistryServiceTest {
     }
 
     @ParameterizedTest
-    @ValueSource(booleans = {false, true})
+    @ValueSource(booleans = { false, true })
     void pullImageAutopullAlways(boolean hasImage) throws Exception {
         givenAnImage();
         givenAnImageConfiguration("myregistry.com/user/app:1.0.1");
@@ -262,34 +268,46 @@ class RegistryServiceTest {
 
     @Test
     void pushBuildXImage() throws MojoExecutionException {
-        givenBuildxImageConfiguration("user/test:1.0.1", null, null);
+        givenBuildxImageConfiguration("user/test:1.0.1", null, null, null);
         givenCredentials("skroob", "12345");
 
         whenPushImage();
 
-        thenBuildxImageHasBeenPushed(null, null);
+        thenBuildxImageHasBeenPushed(null, null, false);
         thenNoExceptionThrown();
     }
 
     @Test
     void pushBuildXImageWithDockerfile() throws MojoExecutionException {
-        givenBuildxImageConfiguration("user/test:1.0.1", null, projectBaseDir.toPath().resolve("src/docker/Dockerfile").toString());
+        String dockerFile = projectBaseDir.toPath().resolve("src/docker/Dockerfile").toString();
+        givenBuildxImageConfiguration("user/test:1.0.1", null, dockerFile, null);
         givenCredentials("skroob", "12345");
 
         whenPushImage();
 
-        thenBuildxImageHasBeenPushed(null, "Dockerfile");
+        thenBuildxImageHasBeenPushed(null, "Dockerfile", false);
         thenNoExceptionThrown();
     }
 
     @Test
     void pushBuildXImageProvidedBuilder() throws MojoExecutionException {
-        givenBuildxImageConfiguration("user/test:1.0.1", "provided-builder", null);
+        givenBuildxImageConfiguration("user/test:1.0.1", "provided-builder", null, null);
         givenCredentials("King_Roland_of_Druidia", "12345");
 
         whenPushImage();
 
-        thenBuildxImageHasBeenPushed("provided-builder", null);
+        thenBuildxImageHasBeenPushed("provided-builder", null, false);
+        thenNoExceptionThrown();
+    }
+
+    @Test
+    void pushBuildXImageTag() throws MojoExecutionException {
+        givenBuildxImageConfiguration("user/test:1.0.1", null, null, "perri-air");
+        givenCredentials("King_Roland_of_Druidia", "12345");
+
+        whenPushImage();
+
+        thenBuildxImageHasBeenPushed(null, null, true);
         thenNoExceptionThrown();
     }
 
@@ -332,17 +350,28 @@ class RegistryServiceTest {
         Mockito.verify(docker).pushImage(Mockito.anyString(), Mockito.any(), Mockito.any(), Mockito.anyInt());
     }
 
-    private void thenBuildxImageHasBeenPushed(String providedBuilder, String relativeDockerfile) throws MojoExecutionException {
+    private void thenBuildxImageHasBeenPushed(String providedBuilder, String relativeDockerfile, boolean tag) throws MojoExecutionException {
         Path buildPath = projectBaseDir.toPath().resolve("target/docker").resolve("user/test/1.0.1");
         String config = getOsDependentBuild(buildPath, "docker");
         String cacheDir = getOsDependentBuild(buildPath, "cache");
         String buildDir = getOsDependentBuild(buildPath, "build");
-        String builderName = providedBuilder!=null ? providedBuilder : "dmp_user_test_1.0.1";
+        String builderName = providedBuilder != null ? providedBuilder : "dmp_user_test_1.0.1";
 
         if (providedBuilder == null) {
             Mockito.verify(exec).process(Arrays.asList("docker", "--config", config, "buildx"),
                 "create", "--driver", "docker-container", "--name", builderName);
         }
+
+        List<String> args = new ArrayList<>();
+        args.addAll(Arrays.asList(
+            "docker", "--config", config, "buildx",
+            "build", "--progress=plain", "--builder", builderName,
+            "--platform", "linux/amd64,linux/arm64"));
+        if (tag) {
+            args.addAll(Arrays.asList("--tag", "user/test:perri-air"));
+        }
+        args.addAll(Arrays.asList("--tag", "user/test:1.0.1", "--push",
+            "--cache-to=type=local,dest=" + cacheDir, "--cache-from=type=local,src=" + cacheDir));
 
         String[] cmds;
         if (relativeDockerfile != null) {
@@ -351,20 +380,13 @@ class RegistryServiceTest {
         } else {
             cmds = new String[] { buildDir };
         }
-        Mockito.verify(exec).process(Arrays.asList("docker", "--config", config, "buildx",
-            "build", "--progress=plain", "--builder", builderName,
-            "--platform", "linux/amd64,linux/arm64",
-            "--tag", "user/test:1.0.1", "--push",
-            "--cache-to=type=local,dest=" + cacheDir, "--cache-from=type=local,src=" + cacheDir), cmds);
+
+        Mockito.verify(exec).process(args, cmds);
 
         if (providedBuilder == null) {
             Mockito.verify(exec).process(Arrays.asList("docker", "--config", config, "buildx"),
                 "rm", builderName);
         }
-    }
-
-    private static String getOsDependentBuild(Path buildPath, String docker) {
-        return buildPath.resolve(docker).toString().replace('/', File.separatorChar);
     }
 
     private void thenImageHasBeenTagged() throws DockerAccessException {
@@ -406,12 +428,12 @@ class RegistryServiceTest {
 
     private void whenPushImage() {
         try {
-             ProjectPaths projectPaths= new ProjectPaths(projectBaseDir, "target/docker");
+            ProjectPaths projectPaths = new ProjectPaths(projectBaseDir, "target/docker");
 
             RegistryService.RegistryConfig registryConfig =
-                    new RegistryService.RegistryConfig.Builder()
-                            .authConfigFactory(authConfigFactory)
-                            .authConfig(authConfig).build();
+                new RegistryService.RegistryConfig.Builder()
+                    .authConfigFactory(authConfigFactory)
+                    .authConfig(authConfig).build();
             registryService.pushImages(projectPaths, Collections.singleton(imageConfiguration), 1, registryConfig, false);
         } catch (Exception e) {
             this.actualException = e;
@@ -447,19 +469,20 @@ class RegistryServiceTest {
     }
 
     private void givenAnImageConfiguration(String imageName) {
-        givenImageNameAndBuildX(imageName, null, null);
+        givenImageNameAndBuildX(imageName, null, null, null);
     }
 
-    private void givenBuildxImageConfiguration(String imageName, String builderName, String dockerFile) {
+    private void givenBuildxImageConfiguration(String imageName, String builderName, String dockerFile, String tag) {
         BuildXConfiguration buildx = new BuildXConfiguration.Builder()
             .platforms(Arrays.asList("linux/amd64", "linux/arm64"))
             .builderName(builderName)
             .build();
-        givenImageNameAndBuildX(imageName, buildx, dockerFile);
+        givenImageNameAndBuildX(imageName, buildx, dockerFile, tag);
     }
 
-    private void givenImageNameAndBuildX(String imageName, BuildXConfiguration buildx, String dockerFile) {
-        BuildImageConfiguration buildImageConfiguration = new BuildImageConfiguration.Builder().buildx(buildx).dockerFile(dockerFile).build();
+    private void givenImageNameAndBuildX(String imageName, BuildXConfiguration buildx, String dockerFile, String tag) {
+        List<String> tags = tag != null ? Collections.singletonList(tag) : null;
+        BuildImageConfiguration buildImageConfiguration = new BuildImageConfiguration.Builder().buildx(buildx).tags(tags).dockerFile(dockerFile).build();
         buildImageConfiguration.initAndValidate(logger);
         imageConfiguration = new ImageConfiguration.Builder().name(imageName).buildConfig(buildImageConfiguration).build();
     }

